src/types/inc/utils.hpp

```diff
@@ -85,4 +85,33 @@ namespace Microsoft::Console::Utils
     }
 
     GUID CreateV5Uuid(const GUID& namespaceGuid, const gsl::span<const gsl::byte> name);
+
+    // Method Description:
+    // - Base case provided to handle the last argument to CoalesceOptionals<T...>()
+    template<typename T>
+    T CoalesceOptionals(const T& base)
+    {
+        return base;
+    }
+
+    // Method Description:
+    // - Base case provided to throw an assertion if you call CoalesceOptionals(opt, opt, opt)
+    template<typename T>
+    T CoalesceOptionals(const std::optional<T>& base)
+    {
+        static_assert(false, "CoalesceOptionals must be passed a base non-optional value to be used if all optionals are empty");
+        return T{};
+    }
+
+    // Method Description:
+    // - Returns the value from the first populated optional, or a base value if none were populated.

```

Comment:
it's pretty much `opt.value_or` without having to write..


```
opt1.value_or(opt2.value_or(opt3.value_or(base)));
```

(instead:

```
CoalesceOptionals(opt1, opt2, opt3, base)
```

)